openshift-knative/serverless-operator
# [WIP] Set owner refs for KafkaChannel resources

Bases https://github.com/openshift-knative/serverless-operator/pull/487
I didn't want to wait until the CI is green and that's merged. This is a WIP PR to see in advance if things fail and if possibly get some reviews.

This PR installs sets ownerRefs on KafkaChannel resources.
KafkaSource resources are put in another namespace, `knative-sources`, anyway and we need to delete stuff manually there.

Deletion of cluster-scoped KafkaChannel resources are also not done yet. To be deleted manually in a separate PR.
There are separate items for these in the umbrella ticket: https://github.com/openshift-knative/serverless-operator/issues/486

As written in https://github.com/openshift-knative/serverless-operator/issues/486, there's still a lot to do.

Verification:

1. Install eventing

2. Enable the controller by uncommenting the stuff in `knative-operator/pkg/controller/add_knativekafka.go`

3. Need to manually create the CRD since it is not in the CSV yet

```
kubectl apply -f knative-operator/deploy/crds/operator_v1alpha1_knativekafka_crd.yaml
```

4. Install:
```
cat <<EOS |kubectl apply -f -
---
apiVersion: operator.serverless.openshift.io/v1alpha1
kind: KnativeKafka
metadata:
  name: example
  namespace: knative-eventing
spec:
  source:
    enabled: true
  channel:
    enabled: true
    bootstrapServers: foo.example.com:1234
...
EOS
```

5. Delete the CR and check if namespaced KafkaChannel related resources are deleted:
```
kubectl delete knativekafkas example -n knative-eventing
```

6. Clean up (operator doesn't delete cluster scoped stuff of KafkaChannel and nothing of KafkaSource at the moment):
```
k delete -f knative-operator/deploy/resources/knativekafka/kafkasource-latest.yaml
k delete -f knative-operator/deploy/resources/knativekafka/kafkachannel-latest.yaml
```
